cli/fossilize_replay.cpp

```diff
@@ -2490,47 +2496,22 @@ struct ThreadedReplayer : StateCreatorInterface
 		}
 	}
 
-	template <typename DerivedInfo>
-	bool derived_work_item_is_satisfied_base(const DerivedInfo &info,
-	                                         const unordered_map<Hash, VkPipeline> &pipelines) const
-	{
-		if (!info.info)
-			return false;
-		auto *library = find_pnext<VkPipelineLibraryCreateInfoKHR>(
-				VK_STRUCTURE_TYPE_PIPELINE_LIBRARY_CREATE_INFO_KHR, info.info->pNext);
-		return !library || pipeline_library_info_is_satisfied(*library, pipelines);
-	}
-
 	template <typename DerivedInfo>
 	bool derived_work_item_is_satisfied(const DerivedInfo &info,
-										const unordered_map<Hash, VkPipeline> &pipelines) const
-	{
-		return derived_work_item_is_satisfied_base(info, pipelines);
-	}
-
-	bool derived_work_item_is_satisfied(const DeferredRayTracingInfo &info,
 	                                    const unordered_map<Hash, VkPipeline> &pipelines) const
 	{
-		if (!derived_work_item_is_satisfied_base(info, pipelines))
-			return false;
-		return !info.info->pLibraryInfo || pipeline_library_info_is_satisfied(*info.info->pLibraryInfo, pipelines);
+		auto *library = work_item_get_library_info(info);
```

Comment:
This loses the condition that if info.info is nullptr (from work_item_is_derived), false is returned. It will instead return true here.

Reply:
`work_item_get_library_info` already returns `nullptr` if `info.info` is `nullptr`.

Reply:
Yes, it does, but then you check if `!library`, which becomes true.

Reply:
oh